java/src/main/java/vowpalWabbit/learner/VWLearner.java

```diff
@@ -10,4 +11,9 @@
 
     // This tells the implementations that an IOException cannot be thrown.
     @Override void close();
+
+    void saveModel(File filename);
+    
+    // Get the JNI pointer for the learner
+    public long getNativePointer();
```

Comment:
Why is this needed?  In response to @JohnLangford's question.  The reason the build is failing is that the function `long getNativePointer()` is defined in this interface (which every typed learner uses) and then not implemented anywhere.  If this is removed it should compile fine.